examples/md-flexible/Simulation.h

```diff
@@ -0,0 +1,469 @@
+   * -do the output each timestep
+   * -collects the duration of every Calculation(Position,Force,Velocity)
+   */
+  void simulate();
+
+  /**
+   * Prints statistics like duration of calculation etc of the Simulation.
+   */
+  void printStatistics();
+
+  /**
+   * Getter for ParticlePropertiesLibrary of Simulation
+   * @return unique_prt(ParticlePropertiesLibrary)
+   */
+  const std::unique_ptr<ParticlePropertiesLibrary<double, size_t>> &getPpl() const;
+
+ private:
+  autopas::AutoPas<Particle, ParticleCell> _autopas;
+  std::shared_ptr<MDFlexConfig> _config;
+  std::ofstream _logFile;
+  std::unique_ptr<ParticlePropertiesLibrary<double, size_t>> _particlePropertiesLibrary;
+  std::unique_ptr<Thermostat<decltype(_autopas), std::remove_reference_t<decltype(*_particlePropertiesLibrary)>>>
+      _thermostat;
+  std::unique_ptr<
+      TimeDiscretization<decltype(_autopas), std::remove_reference_t<decltype(*_particlePropertiesLibrary)>>>
+      _timeDiscretization;
+
+  struct timers {
+    autopas::utils::Timer positionUpdate, forceUpdateTotal, forceUpdateTuning, forceUpdateNonTuning, velocityUpdate,
+        simulate, vtk, init, total, thermostat, boundaries;
+  } _timers;
+
+  /**
+   * Precision of floating point numbers printed.
+   */
+  constexpr static auto _floatPrecision = 3;
+
+  /**
+   * Convert a time and a name to a propperly formatted string.
+   * @param name incl. offset.
+   * @param timeMS in microseconds.
+   * @param numberWidth Width to which the time should be offset.
+   * @param maxTime if passed the percentage of timeMS of maxTime is appended.
+   * @return formatted std::string
+   */
+  std::string timerToString(std::string name, long timeMS, size_t numberWidth = 0, long maxTime = 0);
+};
+
+template <typename Particle, typename ParticleCell>
+void Simulation<Particle, ParticleCell>::initializeParticlePropertiesLibrary() {
+  if (_config->epsilonMap.empty()) {
+    throw std::runtime_error("No properties found in particle properties library!");
+  }
+
+  if (_config->epsilonMap.size() != _config->sigmaMap.size() or _config->epsilonMap.size() != _config->massMap.size()) {
+    throw std::runtime_error("Number of particle properties differ!");
+  }
+
+  _particlePropertiesLibrary = std::make_unique<std::remove_reference_t<decltype(*_particlePropertiesLibrary)>>();
+
+  for (auto [type, epsilon] : _config->epsilonMap) {
+    _particlePropertiesLibrary->addType(type, epsilon, _config->sigmaMap.at(type), _config->massMap.at(type));
+  }
+}
+
+template <class Particle, class ParticleCell>
+void Simulation<Particle, ParticleCell>::initialize(const MDFlexConfig &mdFlexConfig) {
+  _timers.init.start();
+
+  _config = std::make_shared<MDFlexConfig>(mdFlexConfig);
+  initializeParticlePropertiesLibrary();
+  if (_config->deltaT != 0) {
+    _timeDiscretization = std::make_unique<
+        TimeDiscretization<decltype(_autopas), std::remove_reference_t<decltype(*_particlePropertiesLibrary)>>>(
```

Comment:
use type of `_timeDiscretization` here:
```suggestion
        decltype(_timeDiscretization)::element_type(
```